pkg/reconciler/service/install.go

```diff
@@ -25,6 +26,36 @@ type Operation interface {
 	Invoke(ctx context.Context, chartProvider chart.Provider, model *reconciler.Task, kubeClient kubernetes.Client) error
 }
 
+//go:generate mockery --name=ManifestLookup --output=mocks --outpkg=mocks --case=underscore
+type ManifestLookup interface {
```

Comment:
Do this interface and the `Lookup()` function need to be part of the install type?

This looks like specific logic for your component. The reconciler service should stay as simple as possible and only contain logic that is used by the base reconciler or all reconcilers equally.

Also if you have all this in the `connectivityproxy` package, you will only need review from your team.

Reply:
I believe that it might be treated as a simple by universal way for searching for a specific component from in rendered manifest. Because it does not contain anything related to connectivity-proxy I would leave it as is. I would change invalid log message.

Reply:
Yes this could be a general utility to search in the manifest, although I am concerned about the performance of it.

The function does a double conversion `unstructured` -> `manifest` -> `unstructured` which is very expensive (gets done later again to apply the resources). So `renderManifest` gets called twice.

Maybe there would be a better way by having `Lookup()` receive the finished manifest already instead of the unstructured map.

Then this could be called between `renderManifest` and `Apply`.

I think the idea is good but needs some more thinking on how to fit it into the architecture.

Reply:
Hey! Let me jump in. What do you think about the solution I just pushed? There are no double conversions, no huge new functionalities in the `install.go`, just a bonus optional condition. I didn't want to move it straight to the `kubernetes` package as we do not operate on kubernetes resources here but only on manifests. **Didn't check it manually yet**, but I'm finishing my work for today and I wanted to know if this is an ok way to go. What do you guys think, @clebs @tobiscr ?